### PR TITLE
Program: Update with new stake program changes

### PR DESCRIPTION
**Pull request by buffalojoec**
#### Problem

Now that the refactor of the Stake program - to support stakers and validators - is almost complete, the Governance program can be updated to use the new helpers and state.

#### Summary of Changes

Update the dependency, and factor in the changes!

### CODE REVIEW

File: program/Cargo.toml [hunk 1]
```diff
@@ -14,14 +14,14 @@ bytemuck = { version = "1.16.1", features = ["derive"] }
...
+solana-program = "2.0"
 spl-discriminator = "0.3.0"
 spl-program-error = "0.5.0"
 
 [dev-dependencies]
 proptest = "1.5.0"
-solana-program-test = "2.0.2"
-solana-sdk = "2.0.2"
+solana-program-test = "2.0"
+solana-sdk = "2.0"
```
Comment (joncinque) [inline program/Cargo.toml:24]: Were these changes intentional?

**Review by joncinque — APPROVED** [review]
Looks great! Just one comment about the dependency change

File: program/Cargo.toml (hunk 1 above)
Comment (buffalojoec) [reply program/Cargo.toml:24]: Yeah, We have `spl-pod` dependent on `swolana-program = "2.0.3"` in the stake program, and that means anything below that won't be tolerated by Cargo. So to avoid any other dependency issue, I let Cargo sort out the patch versions here.
